src/command_modules/azure-cli-redis/azure/cli/command_modules/redis/custom.py

```diff
@@ -19,9 +19,19 @@ def cli_redis_import_method(client, resource_group_name, name, file_format, file
     parameters = ImportRDBParameters(files, file_format)
     return client.import_method(resource_group_name, name, files, parameters)
 
-def cli_redis_update_settings(client, resource_group_name, name, redis_configuration):
+def cli_redis_update_settings(client, resource_group_name, name, redis_configuration=None, # pylint:disable=too-many-arguments
+                              sku_name=None, sku_family=None, sku_capacity=None):
+    if sku_name != None or sku_family != None or sku_capacity != None:
```

Comment:
It would. Also, looking into it, apparently the "redis-configuration" parameter only lets you change a very specific subset of things that aren't clearly enumerated.

In that case, my recommendation is:
- create a new command called "redis update" that using the "cli_generic_update" pattern that other resource types use.
- Move the SKU related parameters as the convenience layer on that command, and leaving "update-settings" as is. 
- In the future, we can deprecate "update-settings" (as the generic update would accomplish everything it does and more).